homeassistant/components/sensor/device_condition.py

```diff
@@ -116,18 +115,12 @@ async def async_get_conditions(
     ]
 
     for entry in entries:
-        device_class = DEVICE_CLASS_NONE
-        state = hass.states.get(entry.entity_id)
-        unit_of_measurement = (
-            state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) if state else None
-        )
+        device_class = get_device_class(hass, entry.entity_id) or DEVICE_CLASS_NONE
+        unit_of_measurement = get_unit_of_measurement(hass, entry.entity_id)
 
-        if not state or not unit_of_measurement:
+        if not unit_of_measurement:
```

Comment:
Interesting; Should this be:

```suggestion
        if unit_of_measurement is None:
```

Reasoning: A unit of measurement `""` is a valid unit for measurements @ Home Assistant and often used to "hide" a unit, but still make graphs shown and such.

Reply:
I thought the unit of measurement hack is to set it to `" "`, not `""`?
If frontend shows a graph for `""`, the condition should be changed.

Reply:
You are right, it needs to contain at least a space for that 👍